### PR TITLE
Misc improvements

### PR DESCRIPTION
**Pull request by ngerakines**
- Fixed a bug where query string params were being added as part of the signature.
- Added json string to request when payload given.
- Removed pipes (temporarily)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,14 +17,14 @@ function parsePayload(input) {
...
 
-
 this.config = function (config) {
   if (config) { this.config = merge(this.config,config); }
   return this.config;
 };
 
 this.sign = function (method, resource, date) {
-    var str     = this.config.secret + '\n' + date + '\n' + method + '\n' + resource + '\n',
+    var parsedUrl = url.parse(resource);
```
Comment (ngerakines) [inline index.js:26]: I didn't see any reference to error checking on this. I'm assuming an exception is raised.

File: index.js [hunk 2]
```diff
@@ -42,19 +45,27 @@ this.raw = function (method, endpoint, payload, callback) {
           'date'         : date
         },
         options = { url: baseURI + endpoint, method: method, headers: headers };
-    
-    request(options)
```
Comment (ngerakines) [inline index.js:46]: The cln api has resources return objects and others arrays which meant having to add some funky collector logic, so I removed streaming.

File: index.js [hunk 3]
```diff
@@ -42,19 +45,27 @@ this.raw = function (method, endpoint, payload, callback) {
...
-      callback(undefined,data);
-    }));
+
+        if (arguments.length === 4) {
+            options.body = JSON.stringify(payload)
+        }
+
+    request(options, function (error, response, body) {
+        if (!error) {
+            if (response.statusCode == 204) {
```
Comment (ngerakines) [inline index.js:55]: When a 204 (no content) is returned, simply return the status code and a null body. For all other cases, including errors, a json structure should be parsed.

File: test/checklist-ninja.spec.js [hunk 4]
```diff
@@ -12,5 +11,9 @@ describe("Checklist Ninja", function () {
     expect(config).to.have.deep.property('pubkey','bar');
   });
 
+  it("sign", function() {
+  	var signature = ninja.sign('GET', '/path/to?key=value', 'date')
+    expect(signature).to.equal('ok');
```
Comment (ngerakines) [inline test/checklist-ninja.spec.js:16]: This test fails.

File: index.js (hunk 2 above)
Comment (markhuge) [inline index.js:46]: Reminder for myself to look at this.

**Pull request merged**